File: machi_koro/cards/establishment_base.py
from dataclasses import dataclass
from machi_koro.card_deck import CardDeck
from machi_koro.cards.base_card import BaseCard
from machi_koro.card_effects.establishments.establishment_effect_base import CardEffect
from machi_koro.utils import CardIcon


@dataclass
class EstablishmentBase(BaseCard):
    card_icon: CardIcon = CardIcon.UNKNOWN
    construction_cost: int = None
    activation_number: list() = None
    card_effect: CardEffect = None
    deck: CardDeck = None

File: MachiKoro.Core/Models/Cards/Establishments/Basic/EstablishmentBase.cs
using MachiKoro.Core.CardEffects.Establishments.Basic;
using System.Collections.Generic;

namespace MachiKoro.Core.Cards.Establishments.Basic
{
    public abstract class EstablishmentBase : MachiKoro.Core.Cards.ICard
    {
        private readonly IEstablishmentEffect _establishmentEffect;
        public string Name { get; }
        public CardCategory CardIcon { get; }
        public List<int> ActivationNumbers { get; }
        public int ConstructionCost { get; }
       
        public EstablishmentBase(string name, CardCategory cardIcon, List<int> activationNumbers, int constructionCost, IEstablishmentEffect establishmentEffect)
        {
            Name = name;
            CardIcon = cardIcon;
            ActivationNumbers = activationNumbers;
            ConstructionCost = constructionCost;
            _establishmentEffect = establishmentEffect;
        }

        public void Activate(IGame game)
        {
            _establishmentEffect.Activate(game);
        }
    }
}